src/components/video.tsx

```diff
@@ -109,15 +109,22 @@ export function getVideoProps(allProps: VideoProps, state: { asset?: Asset }) {
 
   if (asset) {
     if (asset.status === 'ready') {
-      props.blurDataURL = blurDataURL ?? asset.blurDataURL;
+      props.blurDataURL ??= asset.blurDataURL;
 
-      // Mux provider logic
+      // Mux provider
       const playbackId = asset.externalIds?.playbackId;
 
       if (playbackId) {
-        props.src ??= `https://stream.mux.com/${playbackId}.m3u8`;
+        // src can't be overridden by the user.
+        props.src = `https://stream.mux.com/${playbackId}.m3u8`;
         props.poster ??= getPosterURLFromPlaybackId(playbackId, props);
       }
+      // Vercel Blob provider
+      else if (asset.externalIds?.url) {
```

Comment:
This is good for getting this working, but doesn't really provide a path forward for adding other providers. No change needed for now.

Reply:
yep agreed